From 49cc6f9c79195d243f0e8d035267d9734708e5a5 Mon Sep 17 00:00:00 2001
From: Dmitry Timoshkov <dmitry@baikal.ru>
Date: Wed, 6 Apr 2016 15:14:25 +0800
Subject: [PATCH] user32: Before asking a WM to activate a window make sure
 that the window is in foreground and not minimized.

This patch fixes iconify action using WM's taskbar buttons for Winamp.
---
 dlls/win32u/input.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/dlls/win32u/input.c b/dlls/win32u/input.c
index ade8c313423..dda2a750275 100644
--- a/dlls/win32u/input.c
+++ b/dlls/win32u/input.c
@@ -1377,6 +1377,10 @@ static BOOL set_active_window( HWND hwnd, HWND *prev, BOOL mouse, BOOL focus )
                       (LPARAM)previous );
         if (NtUserGetAncestor( hwnd, GA_PARENT ) == get_desktop_window())
             NtUserPostMessage( get_desktop_window(), WM_PARENTNOTIFY, WM_NCACTIVATE, (LPARAM)hwnd );
+
+        if (hwnd == NtUserGetForegroundWindow() && !is_iconic( hwnd ))
+            NtUserSetActiveWindow( hwnd );
+
     }
 
     /* now change focus if necessary */
-- 
2.35.1

